File: labs/lab_14.py
def pick6():
    import random
    # Return 6 random numbers in a cipher_list
    my_ticket = []
    for count in range(6):
        my_ticket.append(random.randint(1, 100))

    return my_ticket

def matches(winning_ticket, my_ticket):
    counter = 0

    for i in range(len(winning_ticket)):
        if winning_ticket[i] == my_ticket[i]:
            counter = counter + 1

    if counter == 0:
        return 0
    if counter == 1:
        return 4
    if counter == 2:
        return 7
    if counter == 3:
        return 100
    if counter == 4:
        return 50000
    if counter == 5:
        return 1000000
    if counter == 6:
        return 25000000

def play_pick6():
    spend = 0
    winnings = 0
    winning_ticket = pick6()

    for count in range(100000):
        my_ticket = pick6()
        winnings = winnings + matches(winning_ticket, my_ticket)
        spend = spend + 2

    roi = winnings - spend

    print(f"You spent ${spend}\nYou won ${winnings}\nROI = ${roi}")

play_pick6()

File: labs/js labs/basic/lab14.js
function pick6 () {
    let ticket = [];
    for (let i = 0; i < 6; i++) {
        ticket[i] = Math.floor((Math.random() * 100) + 1)
    }
    return ticket
}

function matches (winningTicket, myTicket) {
    let counter = 0;

    for (let i = 0; i < winningTicket.length; i++) {
        if (winningTicket[i] === myTicket[i]) {
            counter += 1;
        }
    }
    if (counter === 0) {
        return 0
    }
    if (counter === 1) {
        console.log('Match 1: '+ myTicket);
        return 4
    }
    if (counter === 2) {
        console.log('Match 2 '+ myTicket);
        return 7
    }
    if (counter === 3) {
        console.log('Match 3: '+ myTicket);
        return 100
    }
    if (counter === 4) {
        console.log('Match 4: '+ myTicket);
        return 50000
    }
    if (counter === 5) {
        console.log('Match 5: '+ myTicket);
        return 1000000
    }
    if (counter === 6) {
        console.log('Match 6: '+ myTicket);
        return 25000000
    }
}

function playGame() {
    let spend = 0;
    let winnings = 0;
    let winner = pick6();
    console.log('Winning Ticket: ' + winner);

    for (let i = 0;i < 10; i++) {
        let my_ticket = pick6();
        winnings += matches(winner, my_ticket);
        spend += 2;
    }

    let result = ('You spent $' + spend + '. \nYou won $' + winnings + '.\n')
    return result
}

let result = playGame()

console.log(result)